### PR TITLE
Maint-23712 : fix Notification sent when task's status is modified

### PR DESCRIPTION
**Pull request by ahmadmido**
This PR will fix the current issue : 
When the assignee and coworkers change the task's status the creator receives a notification.

### CODE REVIEW

File: integration/src/main/java/org/exoplatform/task/integration/notification/TaskEditionPlugin.java [hunk 1]
```diff
@@ -37,6 +37,6 @@ public String getId() {
   public boolean isValid(NotificationContext ctx) {
     Task task = ctx.value(NotificationUtils.TASK);
     return((task.getAssignee() != null && !task.getAssignee().isEmpty()) ||
-            (task.getCoworker() != null && task.getCoworker().size() > 0)||(task.getWatcher() != null && task.getWatcher().size() > 0));
+            (task.getCoworker() != null && task.getCoworker().size() > 0)||(task.getWatcher() != null && task.getWatcher().size() > 0) || (task.getCreatedBy() != null && task.getCreatedBy() != null));
```
Comment (ahamdi) [inline integration/src/main/java/org/exoplatform/task/integration/notification/TaskEditionPlugin.java:40]: There is an unneeded check for nullity for task.getCreatedBy()
```suggestion
            (task.getCoworker() != null && task.getCoworker().size() > 0)||(task.getWatcher() != null && task.getWatcher().size() > 0) || (task.getCreatedBy() != null && task.getCreatedBy() != null));
```

File: integration/src/test/java/org/exoplatform/task/integration/notification/AbstractNotificationPluginTest.java [hunk 2]
```diff
@@ -51,6 +51,23 @@ public void shouldReturnAssigneeInReceivers() throws Exception {
     assertEquals("user1", receivers.iterator().next());
   }
 
+  @Test
+  public void shouldReturnCreatorOfTaskInReceivers() throws Exception {
```
Comment (ahamdi) [inline integration/src/test/java/org/exoplatform/task/integration/notification/AbstractNotificationPluginTest.java:55]: Creator should not receive a notification when a task is created. Please check the test function in this file shouldNotReturnCreatorInReceivers()

Comment (ahmadmido) [reply integration/src/test/java/org/exoplatform/task/integration/notification/AbstractNotificationPluginTest.java:55]: shouldNotReturnCreatorInReceivers() this function check that the notification creator shouldn't receive notification, but shouldReturnCreatorOfTaskInReceivers() function check if creator of task receive notification .

Comment (ahamdi) [reply integration/src/test/java/org/exoplatform/task/integration/notification/AbstractNotificationPluginTest.java:55]: Yes, my question was more about adding the creator or task to the list of receiver when we edit the task and not when he creates it. @mzorai WDYT ?

Comment (ahmadmido) [reply integration/src/test/java/org/exoplatform/task/integration/notification/AbstractNotificationPluginTest.java:55]: No one will receive a notification when the task is created. Only when the task is edited, all concerned will receive the notification (watchers, coworkers, creator, assignee)

Comment (ahamdi) [reply integration/src/test/java/org/exoplatform/task/integration/notification/AbstractNotificationPluginTest.java:55]: Ok good then

**Pull request merged**